client/me/help/help-contact/index.jsx

```diff
@@ -666,7 +666,7 @@ const HelpContact = React.createClass( {
 				<Card className={ this.canShowChatbox() ? 'help-contact__chat-form' : 'help-contact__form' }>
 					{ this.getView() }
 				</Card>
-				<HappychatConnection />
+				{ this.props.shouldStartHappychatConnection && <HappychatConnection /> }
```

Comment:
There are other entry points into Happychat (pre-sales, pre-cancellation). Should those entry points have the same safeguard put in place?

As-is, this leaves the possibility that a customer renders a `<HappychatConnection />` elsewhere in the app, before sites are done being requested (meaning if their primary site is a JPOP site, their HC connection is still connected as a WPCOM chat). When they go to the help form, since a connection has been established the `<HappychatConnection />` won't do anything to reset preferences to the JPOP group. So the bug still remains.

This is admittedly unlikely, but I worry that this and other similar bugs will keep coming back until we come up with a better solution for how every Happychat entry point can have equal access to setting groups. I know we bikeshedded this a lot, but I really think a better long-term solution would be something like [my suggestion](https://github.com/Automattic/wp-calypso/pull/15013#issuecomment-308555339) that `<HappychatConnection />` should require a `groups` prop so that any entry point into Happychat can very clearly say "this is the group this chat should start in".

If this solves an immediate problem without too much hassle, that's fine — but I just wanted to raise this as another issue that might keep biting us.